cmdx.py

```diff
@@ -1781,23 +1795,83 @@ def setLimit(self, typ, value):
 kTranslateMinZ = 22
 
 
-class ObjectSet(Node):
-    """Support set-type operations on Maya sets
+class Mesh(DagNode):
 
-    Caveats
-        1. MFnSet was introduced in Maya 2016, this class backports
-            that behaviour for Maya 2015 SP3
+    _Fn = om.MFnMesh
+    _Its = {
+        om.MFn.kMeshEdgeComponent: om.MItMeshEdge,
+        om.MFn.kMeshFaceVertComponent: om.MItMeshFaceVertex,
+        om.MFn.kMeshPolygonComponent: om.MItMeshPolygon,
+        om.MFn.kMeshVertComponent: om.MItMeshVertex
+    }
 
-        2. Adding a DAG node as a DG node persists its function set
-            such that when you query it, it'll return the name rather
-            than the path.
+    def __init__(self, mobject, *args, **kwargs):
+        super(Mesh, self).__init__(mobject, *args, **kwargs)
+        self._its = {}
 
-            Therefore, when adding a node to an object set, it's important
-            that it is added either a DAG or DG node depending on what it it.
+        self._vtx = None
 
-            This class manages this automatically.
+    def iterator(self, type):
+        """Return mesh iterator of given component type
 
-    """
+        Arguments:
+            type (int): MFn mesh component type
+        """
+        if type not in self._Its:
+            log.warning("invalid iterator type")
+            return
+
+        it = self._its.get(type)
+        if it is None:
+            it = self._Its[type](self.dagPath())
+            self._its[type] = it
+        return it
+
+    @property
+    def numVertices(self):
+        self._fn.syncObject()
+        return self._fn.numVertices
+
+    @property
+    def vtx(self):
+        """Return the full vertex Component object of this mesh
+        """
+        if self._vtx is None:
+            k = om.MFn.kMeshVertComponent
+
+            cps = om.MFnSingleIndexedComponent().create(k)
+            fn = om.MFnSingleIndexedComponent(cps)
+            fn.setCompleteData(self.numVertices)
+
+            self._vtx = MeshVertex(self, cps)
```

Comment:
Will this break, if new vertices are added to a mesh after first reference by cmdx? E.g. (1) `mesh.vtx` is accessed, (2) mesh is subdivided, (3) `mesh.vtx` is accessed again, (4) boom?

Reply:
Let's add some docstring examples here to test this.

Reply:
I added a sync somewhere in the code for this purpose. so the _fn of the components would work even if the geometry changed.

But you're right, I'll add some docstring tests!

Reply:
I suspect the intended usage of this and other functions sets is:

```py
fn = om.MFnSingleIndexedComponent()
cps = fn.create(k)
```

Where `fn` would get the latest created MObject associated with it automatically.

Reply:
I investigated a bit on this one. there is nothing to optimize here. It looks like there fn.create is a sort of classmethod of MFnSingleComponent. caching an void instance of it has not impact on performance.

The way the implemented components in maya is a bit weird. A component is a combination of a node (MDagPath) and a index list with a specified type. So you don't have any direct link between these 2 objects. What i mean is if you want to be sure to not have any problem with getting indices out of range, we have to check that ourselves and that's what i added in the code. So reusing the same casted component MObject doesn't really matter (well that's how i understood it)

And yes I have to add these docstring check (: (and every other missing comments)